api/src/OpenTelemetry/Internal/Logging/Types.hs

```diff
@@ -126,55 +126,139 @@ data LogRecordArguments body = LogRecordArguments
   , observedTimestamp :: Maybe Timestamp
   , context :: Maybe Context
   , severityText :: Maybe Text
-  , severityNumber :: Maybe Int
+  , severityNumber :: Maybe SeverityNumber
   , body :: body
   , attributes :: H.HashMap Text AnyValue
   }
 
 
-data SeverityNumber = SeverityNumber {shortName :: Maybe Text, severityInt :: !Int64} deriving (Read, Show)
+emptyLogRecordArguments :: body -> LogRecordArguments body
+emptyLogRecordArguments body =
+  LogRecordArguments
+    { timestamp = Nothing
+    , observedTimestamp = Nothing
+    , context = Nothing
+    , severityText = Nothing
+    , severityNumber = Nothing
+    , body = body
+    , attributes = H.empty
+    }
+
+
+data SeverityNumber
+  = Trace
+  | Trace2
+  | Trace3
+  | Trace4
+  | Debug
+  | Debug2
+  | Debug3
+  | Debug4
+  | Info
+  | Info2
+  | Info3
+  | Info4
+  | Warn
+  | Warn2
+  | Warn3
+  | Warn4
+  | Err
+  | Err2
+  | Err3
+  | Err4
```

Comment:
`Err` is used because `Error` is taken by another data structure in the haskell ecosystem that was causing naming conflicts. There may be a better name for this but `Err` seems fine.

Reply:
Did some poking around and it actually looks like [the conflict is coming from here](https://opentelemetry.io/docs/specs/otel/trace/api/#set-status). Unfortunate that they used `Error` in two parts of the Spec!

I feel like `SpanStatus` should be able to use `Error` because the spec explicitly uses that term whereas `SeverityNumber` defines the short name as `ERROR` but has no guidance on `Enum` names. It could be qualified, but I feel like that is more cumbersome than naming it `Err`.